core/src/engine/coro/pool.hpp

```diff
@@ -45,6 +49,17 @@ class Pool final {
   const PoolConfig config_;
   const Executor executor_;
 
+  // Maximum thread local coroutine cache size
+  static constexpr std::size_t kLocalCoroutineCacheMaxSize = 32;
+  // Maximum number of coroutines exchanged between
+  // used_coroutines_ for thread local coroutine cache
+  // for sure kLocalCoroutineMoveSize <= kLocalCoroutineCacheMaxSize
+  static constexpr std::size_t kLocalCoroutineMoveSize = 16;
+  static_assert(kLocalCoroutineMoveSize <= kLocalCoroutineCacheMaxSize);
+
+  static inline thread_local std::vector<std::optional<Coroutine>>
```

Comment:
Now, when we use `std::make_move_iterator` and `std::back_inserter`, we may be able to remove `std::optional` again. We'll need to bring back `CoroutineMover` in `GetCoroutine`